server/lib/reverse_etl/utils/cursor_query_builder.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module ReverseEtl
+  module Utils
+    class CursorQueryBuilder
+      def self.update_model_query(sync_config)
+        existing_query = sync_config.model.query
+
+        if sync_config.cursor_field && sync_config.current_cursor_field
+          new_query = "(#{existing_query}) AS subquery " \
+          "WHERE #{sync_config.cursor_field} >= '#{sync_config.current_cursor_field}' " \
+          "ORDER BY #{sync_config.cursor_field} ASC"
```

Comment:
Why do we need `order by`? Won't it make the query expensive?

Reply:
IMHO cursor takes care of the consistency and predictability and order by put load on the Database especially if the cursor field is not indexed. But it's the teams call. But as an exercise we should analyse the query with and with order by for indexed and without indexed cursor field.

Reply:
@afthabvp  Not clear on the why we need to order the results

Reply:
ORDER BY clause adds overhead, especially if the cursor field is not indexed, it's a necessary trade-off for ensuring data integrity and efficient querying. Without ORDER BY, as you mentioned, you would need an additional query to fetch the cursor field value, which introduces unnecessary complexity and overhead.